tools/distrib/upgrade_boringssl_objc.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generate the list of boringssl symbols that need to be shadowed based on the
+# current boringssl submodule. Requires local toolchain to build boringssl.
+
+set -e
+
+cd "$(dirname $0)"
+cd ../../third_party/boringssl
+
+BORINGSSL_COMMIT=$(git rev-parse HEAD)
+BORINGSSL_PREFIX_HEADERS_DIR=src/boringssl
+
+# Do the following in grpc root directory
+cd ../..
+
+docker build tools/dockerfile/grpc_objc/generate_boringssl_prefix_header -t grpc/boringssl_prefix_header
+mkdir -p $BORINGSSL_PREFIX_HEADERS_DIR
+docker run -it --rm -v $(pwd)/$BORINGSSL_PREFIX_HEADERS_DIR:/output grpc/boringssl_prefix_header $BORINGSSL_COMMIT
+
+# Increase the minor version by 1
```

Comment:
I don't like that this script is not idempotent - if you run it multiple times, it will keep increasing the version number.
IMHO just having a robust .sh that regenerates the boringssl_prefix_symbols.h is enough for this script.

If you want to add a script that automates the whole boringssl update process, we can look into that but there are questions to be answered first and it doesn't belong to this PR.  I'd recommend turning this script into something that just generates the header file (with use of docker if you will).